rs/ledger_suite/tests/sm-tests/src/fee_collector.rs

```diff
@@ -251,7 +267,16 @@ where
                     .expect("Unable to perform the transfer");
                 transfer(&env, ledger_id, account_from, account_to, amount)
                     .expect("Unable to perform the transfer");
-                let blocks = get_blocks(&env, ledger_id.get().0, block_id, 3).blocks;
+                let blocks = match block_retrieval {
+                    BlockRetrieval::Legacy => {
+                        get_blocks(&env, ledger_id.get().0, block_id, 3).blocks
+                    }
+                    BlockRetrieval::Icrc3 => icrc3_get_blocks(&env, ledger_id, block_id, 3)
+                        .blocks
+                        .into_iter()
+                        .map(|b| GenericBlock::from(b.block))
+                        .collect(),
+                };
```

Comment:
Although not exactly the same, it seems like mostly what's going on in 226-233. Could we refactor this to decrease complexity in this function?

Reply:
Good idea, done!